xap-extensions/xap-data-gateway/src/main/java/com/gigaspaces/sql/datagateway/netty/server/MessageProcessor.java

```
@@ -368,18 +420,22 @@ private void onParse(ChannelHandlerContext ctx, ByteBuf msg) throws ProtocolExce
             paramTypes[i] = msg.readInt();
         try {
             queryProvider.prepare(session, stmt, query, paramTypes);
-        } catch (ProtocolException e) {
-            throw e;
         } catch (Exception e) {
-            throw new NonBreakingException(ErrorCodes.INTERNAL_ERROR, "cannot prepare statement", e);
+            throw wrapException("Failed to prepare statement", e);
```

I ran regression, and I don't see it failing on such queries
But I see that the exception that is returned from the gateway is masking the error message from the cause during the wrapping which is crucial for cases that the user wants to know why his query failed, for instance when running select with a wrong column name we expect PSQLException error that contains "Column X not found in any table"

Do you have an Idea how to fix it? maybe append all the error messages from the cause to the root exception? like: "Failed to prepare statement due to: 'Column X not found in any table' "

---

Fixed. At now a wrapped exception contains an original message from its cause. Also, I did some cosmetic changes to make validation exceptions more readable